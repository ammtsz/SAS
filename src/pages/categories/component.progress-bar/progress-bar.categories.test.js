import React from "react";
import ProgressBar from "./progress-bar.categories.component";

import { render, screen, cleanup } from "@testing-library/react";

describe("Component categories-page <ProgressBar />", () => {
  const mockProps = {
    progress: 8,
  };

  let component;

  beforeEach(() => {
    component = render(<ProgressBar {...mockProps} />);
  });

  afterEach(() => {
    cleanup();
  });

  it("should render ProgressBar", () => {
    expect(component.asFragment()).toMatchSnapshot();
  });

  it("should print 8 colored units in '<ProgressBar />' component", () => {
    expect(screen.getAllByTestId("progress-bar-colored").length).toBe(8);
  });

});
