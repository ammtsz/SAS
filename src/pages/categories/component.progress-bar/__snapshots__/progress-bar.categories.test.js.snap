// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component categories-page <ProgressBar /> should render ProgressBar 1`] = `
<DocumentFragment>
  <div
    aria-label="8 questions answered"
    class="sc-bdfBwQ ccxwCN progress-bar"
    data-testid="progress-bar"
  >
    <span
      class="sc-gsTCUz dmqapp colored "
      data-testid="progress-bar-colored "
    />
    <span
      class="sc-gsTCUz dmqapp colored "
      data-testid="progress-bar-colored "
    />
    <span
      class="sc-gsTCUz dmqapp colored "
      data-testid="progress-bar-colored "
    />
    <span
      class="sc-gsTCUz dmqapp colored "
      data-testid="progress-bar-colored "
    />
    <span
      class="sc-gsTCUz dmqapp colored "
      data-testid="progress-bar-colored "
    />
    <span
      class="sc-gsTCUz dmqapp colored "
      data-testid="progress-bar-colored "
    />
    <span
      class="sc-gsTCUz dmqapp colored "
      data-testid="progress-bar-colored "
    />
    <span
      class="sc-gsTCUz dmqapp colored "
      data-testid="progress-bar-colored "
    />
    <span
      class="sc-gsTCUz dmqapp  "
      data-testid=" "
    />
    <span
      class="sc-gsTCUz dmqapp  "
      data-testid=" "
    />
  </div>
</DocumentFragment>
`;
